File: shop-app-server/src/main/java/com/medochek/shopapp/domain/BasketRow.java
package com.medochek.shopapp.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonIgnore;

@Table
@Entity
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class BasketRow {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private Integer count;

    @OneToOne
    @JoinColumn(name = "product_id")
    private Product product;

    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "basket_id")
    private Basket basket;
}

File: shop-app-web-user/src/main/web/src/app/models/basketRow.ts
import {Product} from "./product";

export class BasketRow {
    private _id: number;
    private _count: number;
    private _product:Product;

    constructor(data: any) {
        this._id = data.id;
        this._count = data.count;
        this._product = new Product(data.product);
    }

    get id(): number {
        return this._id;
    }

    set id(value: number) {
        this._id = value;
    }

    get count(): number {
        return this._count;
    }

    set count(value: number) {
        this._count = value;
    }

    get product(): Product {
        return this._product;
    }

    set product(value: Product) {
        this._product = value;
    }

    public toObject() {
        return {
            id: this._id,
            count: this._count,
            product: this._product.toObject()
        }
    }
}